src/db.js

```diff
@@ -194,9 +194,16 @@ export async function addDescApp (name, desc) {
  * @returns {Promise<IDeleteRangeResponse>}
  */
 export async function rmApp (app) {
-	await etcd.delete().all().prefix(`${DESC_BASE_URL}/${app}`).exec()
-	await etcd.delete().all().prefix(`${APP_BASE_URL}/${app}`).exec()
-	return etcd.delete().all().prefix(`${SERVICE_BASE_URL}/${app}`).exec()
+	try {
+		let r = await etcd.delete().all().prefix(`${DESC_BASE_URL}/${app}`).exec()
+		r = await etcd.delete().key(`${APP_BASE_URL}/${app}`).exec()
+		if (r.deleted === '0') {
+			throw Error(`Cannot delete app ${app}`)
+		}
+		r = etcd.delete().all().prefix(`${SERVICE_BASE_URL}/${app}`).exec()
```

Comment:
@gutorc92, a couple of questions:

1. Wouldn't the problem also happens to descriptions and IPs? Why are changing the way it works only for apps?

2. Shouldn't you verify for `r.deleted === '0'` **before** deleting descriptions? You may endup with a wrong state by not considering this side effect.

Reply:
@abilioesteves, I forgot to add a ```/``` services to avoid the same error. Also, I removed desc because it is a property on App.

Reply:
got it!